packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -1,30 +1,46 @@
-export type ActionHandler<Action, ActionType> = (
+export type ActionHandler<
+  Action extends ActionConstraint,
+  ActionType = Action['type'],
+> = (
   ...args: ExtractActionParameters<Action, ActionType>
 ) => ExtractActionResponse<Action, ActionType>;
-export type ExtractActionParameters<Action, T> = Action extends {
+
+export type ExtractActionParameters<
+  Action extends ActionConstraint,
+  T = Action['type'],
+> = Action extends {
   type: T;
-  handler: (...args: infer H) => any;
+  handler: (...args: infer H) => unknown;
 }
   ? H
   : never;
```

Comment:
Hmm. We already seem to be checking that `Action` extends `ActionConstraint` by virtue of the conditional type. Does it makes sense to restrict what it needs to be in the type parameters too?

Minimally it seems like we ought to drop `T`, since that's all we really need `Action` to have. We also ought to use a more than one letter type parameter:

``` suggestion
export type ExtractActionParameters<Action> = Action extends {
  handler: (...args: infer Handler) => unknown;
}
  ? Handler
  : never;
```

The same argument applies to all of the types in this section (I won't add comments for them, but I may come back to this).

Reply:
I agree that dropping `T` is probably the way to go with this. I wanted to minimize disruption in case the second parameter was being used somewhere. I'll check whether we can remove the `T` params from all of the types here.

Reply:
Agreed except that `unknown` would be the wrong return type here; we'd still want `any`. We want this to be compatibile with handlers that return any type, not just `unknown`.

Reply:
Ah right, makes sense, thanks

Reply:
The `{Action,Event}Type` params seem to be necessary for type inference to work around action/event types and the methods that accept or return them. Removing just the event types still triggers "implicit any" errors across multiple controllers.

Reply:
Okay, no problem.